### ISSUE TITLE
[Research] Sleep Quality EEG Predictors of Next-Day Cognitive Performance

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#122]
## Research Topic
EEG features during sleep (especially N3 deep sleep) that predict next-day cognitive performance, combined with HRV biomarkers.

## Paper References

1. **"Sleep Brain and Cardiac Activity Predict Cognitive Flexibility and Conceptual Reasoning Using Deep Learning"** — arXiv, May 2025. Key finding: Hybrid CNN-Transformer model (CogPSGFormer) integrates single-channel EEG and ECG with power band features and HRV metrics to predict executive function (PCET scores). Achieved **80.3% accuracy** classifying cognitive levels from sleep signals. Demonstrates that sleep EEG + HRV together are more predictive than either alone.

2. **"Predicting Sleep Quality with Digital Biomarkers and Artificial Neural Networks"** — Frontiers in Psychiatry, July 2025. First work to use continuous HRV to predict next-day sleep quality (WASO). **90.4% binary classification accuracy.** LF/HF ratio emerged as strongest single HRV correlate. Shows continuous wearable HRV can replace polysomnography for quality prediction.

3. **"Machine Learning Identification of Sleep EEG and EOG Biomarkers for Mortality Risk"** — SLEEP (Oxford), Feb 2025. Higher delta power (0.75-3.75 Hz) in N3 sleep correlated with reduced mortality risk. Sleep spindle density in N2 also predictive. Establishes EEG spectral features during deep sleep as long-term health biomarkers.

4. **"Automatic Sleep Stage Classification Using Deep Learning: Signals, Data Representation, and Neural Networks"** — AI Review, 2024. Comprehensive review of DL architectures for sleep staging. Best models achieve >90% on ISRUC/Sleep-EDF. Single-channel EEG sufficient for 5-stage classification.

## Application to NeuralDreamWorkshop

The system already has:
- Sleep staging model (92.98% accuracy on ISRUC)
- Dream detector (97.20% accuracy)
- Apple Health integration (`ml/health/`)

Missing capability: **predicting next-day cognitive performance from overnight EEG features.** This would add a "Sleep Quality Report" that tells users how well they'll perform cognitively based on last night's sleep architecture.

### New files:
- `ml/models/sleep_quality_predictor.py` — SleepQualityPredictor class
- `ml/models/cognitive_forecast.py` — CognitiveForecast class
- `ml/training/train_sleep_quality.py` — training script using Sleep-EDF + ISRUC
- `client/src/pages/sleep-insights.tsx` — morning cognitive forecast dashboard

### Key classes/methods:
```python
class SleepQualityPredictor:
    """Extracts sleep architecture features from overnight EEG recording."""
    
    def extract_sleep_architecture(self, eeg_overnight: np.ndarray, fs: int = 256) -> dict:
        """Returns: n3_duration_min, n3_delta_power, spindle_density,
                    rem_duration_min, sleep_efficiency, awakenings_count,
                    slow_wave_amplitude, sleep_onset_latency"""
    
    def compute_restorative_score(self, architecture: dict) -> float:
        """0-100 score based on N3 depth, continuity, spindle density"""

class CognitiveForecast:
    """Predicts next-day cognitive readiness from sleep features + HRV."""
    
    def predict(self, sleep_features: dict, hrv_features: dict = None) -> dict:
        """Returns: cognitive_readiness (0-100), attention_forecast,
                    memory_forecast, reaction_time_estimate,
                    recommendations: list[str]"""
```

### Key EEG features to extract during N3:
- Delta power (0.5-4 Hz) — the primary restorative marker
- Slow oscillation amplitude (0.5-1 Hz) — memory consolidation
- Sleep spindle density (11-16 Hz bursts in N2) — learning consolidation
- K-complex count and morphology
- N3 continuity (fragmentation index)
- REM density (rapid eye movements per minute of REM)

### HRV features (from Apple Health / Muse PPG):
- RMSSD during N3 — parasympathetic tone
- LF/HF ratio trajectory across sleep cycles
- HRV decline slope across the night

### API endpoints:
- `POST /sleep/upload-overnight` — upload overnight EEG recording
- `GET /sleep/quality-report` — returns sleep architecture + cognitive forecast
- `GET /sleep/trends` — multi-night trend analysis

## Expected Accuracy/Impact
- Sleep architecture extraction: high accuracy (leverages existing 92.98% sleep staging)
- Cognitive performance prediction: ~75-80% (based on CogPSGFormer results)
- HRV integration adds ~5-10% accuracy boost over EEG alone
- User value: actionable morning insight ("Your deep sleep was 15% below your baseline — expect reduced focus today, consider lighter cognitive tasks")

## Complexity Rating: HIGH
- Requires overnight recording infrastructure (long recording buffer, power management)
- Sleep staging must run continuously over 6-8 hours of data
- HRV integration requires Apple Health / Google Fit API pipeline (partially exists)
- Training requires Sleep-EDF or ISRUC dataset processing
- Frontend needs multi-night trend visualization
